components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -156,13 +140,25 @@ func (g *gpsrtk) receiveAndWriteCorrectionData() {
 	defer g.activeBackgroundWorkers.Done()
 	defer g.closeCorrectionWriter()
 
-	err := g.connectToNTRIP()
+	select {
+	case <-g.cancelCtx.Done():
+		return
+	default:
+	}
+
+	err := g.connectAndParseSourceTable()
+	if err != nil {
+		g.err.Set(err)
+		g.logger.Error("unable to parse source table! Giving up on RTK messages")
+		return
+	}
+
+	scanner, err := g.getStream()
 	if err != nil {
 		g.err.Set(err)
-		g.logger.Error("unable to connect to NTRIP stream! Giving up on RTK messages")
+		g.logger.Error("unable to get NTRIP stream! Giving up on RTK messages")
 		return
 	}
```

Comment:
wondering if theres a way for us to only have to call this in one place, rather than both here and in the for loop. fine with leaving as is especially because I have nothing to propose

Reply:
Yeah, I wondered the same thing, but couldn't come up with something good. but if anyone else sees it, I'd sure like to change it!

Reply:
Aha! Take a look at the more recent commits: I think I've come up with something alright.

Reply:
oh the double loop is great!